cmd/digger/main.go

```diff
@@ -1,73 +1,54 @@
 package main
 
 import (
-	"digger/pkg/digger"
-	"digger/pkg/github"
-	"digger/pkg/models"
-	"digger/pkg/utils"
-	"fmt"
-	"os"
-	"strings"
+	"digger/pkg/ci_runner"
+	"digger/pkg/domain"
+	"digger/pkg/lock"
+	"digger/pkg/scm"
+	"digger/pkg/service"
+	"log"
 )
 
 func main() {
-	diggerConfig, err := digger.NewDiggerConfig("")
+	// Retrieve config
+	diggerConf, err := domain.NewDiggerConfig("")
 	if err != nil {
-		fmt.Printf("Failed to read digger config. %s\n", err)
-		os.Exit(1)
+		log.Fatalf("got an error while retrieving digger config: %v", err)
 	}
-	println("Digger config read successfully")
 
-	lock, err := utils.GetLock()
+	// Get CI event
+	ciRunner := ci_runner.Current()
+	event, err := ciRunner.CurrentEvent()
 	if err != nil {
-		fmt.Printf("Failed to create lock provider. %s\n", err)
-		os.Exit(2)
+		log.Fatalf("got an error while retrieving the event: %v", err)
 	}
-	println("Lock provider has been created successfully")
 
-	ghToken := os.Getenv("GITHUB_TOKEN")
-	if ghToken == "" {
-		println("GITHUB_TOKEN is not defined")
-		os.Exit(3)
-	}
-
-	ghContext := os.Getenv("GITHUB_CONTEXT")
-	if ghContext == "" {
-		fmt.Printf("GITHUB_CONTEXT is not defined. %s\n", err)
-		os.Exit(4)
-	}
-
-	parsedGhContext, err := models.GetGitHubContext(ghContext)
+	// Parse event
+	ps := service.NewParser()
```

Comment:
seems like parser is disconnected from CI runner, although I think it's coupled by nature. Maybe it's ci runner that needs to return appropriate parser, or maybe it just needs to return parsed event. 

I cannot imagine that we will parse github event in 2 different ways for example.

Reply:
Absolutely! I agree on that, the CI Runner should be returning the parsed event. Thanks for the suggestion